testing/testing.sh

```diff
@@ -0,0 +1,158 @@
+    docker network create --driver bridge "$NETWORK_NAME" > /dev/null
+}
+
+# Remove docker network
+function remove_network() {
+    docker network remove "$NETWORK_NAME" > /dev/null
+}
+
+# Find and kill any remaining containers attached to the network
+function cleanup() {
+    local containers
+    containers="$(docker ps --quiet --filter network="$NETWORK_NAME")"
+
+    if [ -n "$containers" ]; then
+        echo >&2 -n "==> Killing $(echo -n "$containers" | grep -c '^') orphaned container(s)..."
+        echo "$containers" | xargs docker kill > /dev/null
+        echo >&2 " done."
+    fi
+}
+
+function docker_run() {
+    local container_name="$1"
+    local docker_run_command="$2"
+    local args="$3"
+
+    echo >&2 "===> Starting ${container_name}"
+    if [ "$container_name" = "$MASTER_CONTAINER_NAME" ]; then
+      eval "docker run --rm --detach --network $NETWORK_NAME --name ${container_name} ${docker_run_command} $image_url ${args}"
+    elif [ "$container_name" = "$WORKER_CONTAINER_NAME" ]; then
+      eval "docker run --rm --detach --network $NETWORK_NAME --name ${container_name} ${docker_run_command} $image_url ${args}"
+    else
+      eval "docker run --rm --network $NETWORK_NAME --name ${container_name} ${docker_run_command} $image_url ${args}"
+    fi
+}
+
+function start_spark_master() {
+    docker_run \
+      "$MASTER_CONTAINER_NAME" \
+      "--publish $SPARK_MASTER_WEBUI_HOST_PORT:$SPARK_MASTER_WEBUI_CONTAINER_PORT $1" \
+      "/opt/spark/bin/spark-class org.apache.spark.deploy.master.Master" > /dev/null
+}
+
+function start_spark_worker() {
+    docker_run \
+    "$WORKER_CONTAINER_NAME" \
+    "--publish $SPARK_WORKER_WEBUI_HOST_PORT:$SPARK_WORKER_WEBUI_CONTAINER_PORT $1" \
+    "/opt/spark/bin/spark-class org.apache.spark.deploy.worker.Worker spark://$MASTER_CONTAINER_NAME:$SPARK_MASTER_PORT" > /dev/null
+}
+
+function wait_container_ready() {
+    local container_name="$1"
+    local host_port="$2"
+    i=0
+    echo >&2 "===> Waiting for ${container_name} to be ready..."
+    while true; do
+        i=$((i+1))
+
+        set +e
+
+        curl \
+          --silent \
+          --max-time "$CURL_TIMEOUT" \
+          localhost:"${host_port}" \
+          > /dev/null
+
+        result=$?
+
+        set -e
+
+        if [ "$result" -eq 0 ]; then
+            break
+        fi
+
+        if [ "$i" -gt "$CURL_MAX_TRIES" ]; then
+            echo >&2 "===> \$CURL_MAX_TRIES exceeded waiting for ${container_name} to be ready"
+            return 1
+        fi
+
+        sleep "$CURL_COOLDOWN"
+    done
+
+    echo >&2 "===> ${container_name} is ready."
+}
+
+function run_spark_pi() {
+    docker_run \
+      "$SUBMIT_CONTAINER_NAME" \
+      "$1" \
+      "/opt/spark/bin/spark-submit --master spark://$MASTER_CONTAINER_NAME:$SPARK_MASTER_PORT --class org.apache.spark.examples.SparkPi /opt/spark/examples/jars/spark-examples_${scala_spark_version}.jar 20"
+}
+
+# Run smoke test
+function run_smoke_test() {
```

Comment:
nit:
maybe `run_standalone_test` or `run_smoke_standalone_test`? We will add some more other test besides standalone

Reply:
Rename as `run_smoke_test_in_standalone`